File: specification/specs/x_pack/transform/preview_transform/PreviewTransformRequest.ts
@rest_spec_name('transform.preview_transform')
class PreviewTransformRequest extends RequestBase {
  query_parameters?: {}
  body?: {
    description?: string
    dest?: TransformDestination
    frequency?: Time
    pivot?: TransformPivot
    source?: TransformSource
    sync?: TransformSyncContainer
  }
}

File: output/csharp/XPack/Transform/PreviewTransform/PreviewTransformRequest.cs
using Nest.XPack;
using Nest.CommonOptions;
using Nest.CommonAbstractions;
using System;
using System.Collections.Generic;
using System.Runtime.Serialization;
namespace Nest.XPack {

	public class PreviewTransformRequest : RequestBase {
		
		[DataMember(Name="description")]
		public string Description { get; set; }


		[DataMember(Name="dest")]
		public TransformDestination Dest { get; set; }


		[DataMember(Name="frequency")]
		public Time Frequency { get; set; }


		[DataMember(Name="pivot")]
		public TransformPivot Pivot { get; set; }


		[DataMember(Name="source")]
		public TransformSource Source { get; set; }


		[DataMember(Name="sync")]
		public TransformSyncContainer Sync { get; set; }

	}
}
